### PR TITLE
ntheory: core of a number

### PR DESCRIPTION
**Pull request by postvakje**
Calculate the core of an integer: core(n,t) is the t-th power free part
of n.  core(n,2) is the squarefree part of n.
http://en.wikipedia.org/wiki/Square-free_integer#Squarefree_core

``` python
>>> core(3**6,6) = 1
>>> core(3**7,6) = 3
```

### CODE REVIEW

File: sympy/ntheory/factor_.py [hunk 1]
```diff
@@ -1482,3 +1482,73 @@ def eval(cls, n, k=1):
...
+
+    See Also
+    ========
+
+    factorint
+    """
+
+    @classmethod
+    def eval(cls, n, t=2):
+        n = sympify(n)
```
Comment (smichr) [inline sympy/ntheory/factor_.py]: for speed and simplicity we generally handle this as
```python
n = as_int(n)
t = as_int(t)
```
then
```
if sympify(n).is_prime:
    return n
if n <= 0:
    er
elif t <= 1:
    er
else:
    calc
```

`as_int` will handle raising the error.

File: sympy/ntheory/factor_.py [hunk 2]
```diff
@@ -1482,3 +1482,73 @@ def eval(cls, n, k=1):
...
+
+    .. [1] http://en.wikipedia.org/wiki/Square-free_integer#Squarefree_core
+
+    Examples
+    ========
+
+    >>> from sympy.ntheory.factor_ import core
+    >>> core(24, 2)
+    6
+    >>> core(9424,3)
```
Comment (smichr) [inline sympy/ntheory/factor_.py]: space after comma

**Comment by smichr** [comment]
:+1: This looks good. And good revisions, too. I think making this a function (in light of the fact that you are working with factorint and we don't have an unevaluated `factorint` form).

:information_source: Your code snippets (like the one in the OP) will look better if you paste them between triple back-ticks like this:

\`\`\`python
\>\>\> code snippet
\`\`\`

**which ends up looking like this**
```python
>>> code snippet
```
**instead of this**

>>> code snippet

BTW, you can click on the pencil icon in the upper right hand corner to edit comments, I believe.

**Comment by smichr** [comment]
Let's see if anyone else sees anything and then I'll commit this evening.

**Comment by postvakje** [comment]
Travis failed, but due to possible bugs in other places (see issue #8949).  I'll see if I can fixed those.

File: sympy/ntheory/residue_ntheory.py [hunk 3]
```diff
@@ -577,7 +577,7 @@ def is_quad_residue(a, p):
     prime, an iterative method is used to make the determination:
 
     >>> from sympy.ntheory import is_quad_residue
-    >>> list(set([i**2 % 7 for i in range(7)]))
+    >>> sorted([i**2 % 7 for i in range(7)])
```
Comment (smichr) [inline sympy/ntheory/residue_ntheory.py]: `sorted(set` in both places.

**Comment by smichr** [comment]
:rewind: Could you please just keep the first 4 commits (I'll take care of the rest in #8957)? And please squash those 4 together. :eyeglasses: See also: **Changing of commit messages** at [the wiki](https://github.com/sympy/sympy/wiki/Development-workflow)

**Comment by postvakje** [comment]
Sure, will do it tonight.

**Comment by smichr** [comment]
Thanks.

**Comment by smichr** [comment]
:heavy_plus_sign: This is in.

**Comment by skirpichev** [comment]
> Let's see if anyone else sees anything 

Bah.  Public name like "core" - for sure nobody can see...

**Pull request merged**